File: backend/backend/src/SignalingComponent/Common/DTOs/ErrorMessage.cs
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.DTOs;

namespace backend.SignalingComponent.Common.DTOs;

public class ErrorMessage : ITypedMessage
{
    public static string TypeString => "error";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;

    [JsonPropertyName("requestID")]
    public required string RequestId { get; set; }
    
    [JsonPropertyName("description")]
    public required string Description { get; set; }
    
    [JsonPropertyName("expected")]
    public string? Expected { get; set; }
    
    [JsonPropertyName("actual")]
    public string? Actual { get; set; }
}

File: frontend/src/types/common/ErrorMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class ErrorMessage implements ITypedMessage {
    public readonly type = MessageType.ERROR;
    public msg: {
        requestID: string;
        description: string;
        expected?: string;
        actual?: string;
    };

    public constructor(msg: {
        requestID: string;
        description: string;
        expected?: string;
        actual?: string;
    }) {
        this.msg = msg;
    }
}
